### PR TITLE
Prepare mermaid for diagrams and add example in new structure

### PR DESCRIPTION
**Pull request by margau**

### CODE REVIEW

**Comment by e1mo** [comment]
I can't build it, it fails with the same errors the CI check fails:

```
make  html                     
Running Sphinx v3.4.0

Extension error:
Could not import extension sphinxcontrib.mermaid (exception: No module named 'six')
make: *** [Makefile:20: html] Error 2
```

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 sphinx
 sphinx_rtd_theme
 recommonmark
+sphinxcontrib-mermaid
```
Comment (e1mo) [inline requirements.txt:4]:
```suggestion
sphinxcontrib-mermaid
six
```

This made it build for me

Comment (margau) [reply requirements.txt:4]: Fixed in b2c7a43f008e84ee16baddf6cf42a00189659710

**Review by e1mo — APPROVED** [review]

**Review by adnidor — APPROVED** [review]
lgtm

**Pull request merged**